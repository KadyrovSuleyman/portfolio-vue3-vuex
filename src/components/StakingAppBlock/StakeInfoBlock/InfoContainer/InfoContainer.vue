<script setup lang="ts">

import propsObj from '@/elements/propsObj';
import { computed } from 'vue';
import Div from '@/elements/Div/Div.vue';
import { useStore } from 'vuex';
import adapt from './adapter';
import InfoItem from './InfoItem/InfoItem.vue';

// eslint-disable-next-line no-undef
const props = defineProps({ ...propsObj });
const comp = computed(() => ({ elem: props.elem || 'infoContainer' }));

const store = useStore();
const state = computed(() => adapt(store));

const keys = computed(() => Object.keys(state.value));

</script>

<template>

  <Div :block="props.block" :elem="comp.elem" :mods="props.mods">

    <InfoItem :block="comp.elem" :elem="'infoItem'"
      v-for="key in keys" :key="key"
    >
      <template v-slot:header>
        {{ key }}
      </template>
      {{ state[key] }}
    </InfoItem>

  </Div>

</template>

<style lang="scss">
  @use 'styles' as *;
</style>
