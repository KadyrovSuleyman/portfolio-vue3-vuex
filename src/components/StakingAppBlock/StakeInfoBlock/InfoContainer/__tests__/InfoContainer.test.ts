import { mount, VueWrapper } from '@vue/test-utils';
import { createStore, Store } from 'vuex';
import InfoContainer from '../InfoContainer.vue';

jest.mock('../state');

let store: Store<any>;
beforeEach(() => {
  store = createStore<any>({
    state: {
      From: '',
      To: '',
      Staked: 0,
      APY: '',
    },
    mutations: {
      change: (state, obj: { [name: string]: boolean | string }) => {
        Object.keys(obj).forEach((index) => {
          state[index] = obj[index];
        });
      },
    },
  });
});

let wrapper: VueWrapper<any>;
afterEach(() => {
  wrapper.unmount();
});

it('infoContainer renders', () => {
  wrapper = mount(InfoContainer, {
    global: { plugins: [store] },
  });
  expect(wrapper.element.outerHTML).toMatchSnapshot();
});

it('watchs props changes', async () => {
  wrapper = mount(InfoContainer, {
    global: { plugins: [store] },
  });
  expect(wrapper.find('.infoContainer').classes()).toEqual(['infoContainer']);

  await wrapper.setProps({
    ...wrapper.props,
    mods: {
      selected: true,
    },
  });
  expect(wrapper.find('.infoContainer').classes()).toEqual([
    'infoContainer',
    'infoContainer__selected',
  ]);

  await wrapper.setProps({
    ...wrapper.props,
    mods: {
      selected: false,
    },
  });
  expect(wrapper.find('.infoContainer').classes()).toEqual(['infoContainer']);
});

it('watchs outer store', async () => {
  wrapper = mount(InfoContainer, {
    global: { plugins: [store] },
  });

  const target = wrapper.findAll('.infoContainer-infoItem')[0];
  expect(target.find('.infoItem-value').text()).toBe('');

  store.commit('change', {
    From: 'after',
  });
  await wrapper.vm.$nextTick();
  expect(target.find('.infoItem-value').text()).toBe('after');
});
