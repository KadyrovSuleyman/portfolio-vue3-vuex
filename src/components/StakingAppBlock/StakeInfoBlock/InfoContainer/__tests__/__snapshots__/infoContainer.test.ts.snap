// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`infoContainer renders 1`] = `
<div class="infoContainer">
  <div class="infoContainer-infoItem"><span class="infoItem-header">From</span><span class="infoItem-value"> </span></div>
  <div class="infoContainer-infoItem"><span class="infoItem-header">To</span><span class="infoItem-value"> </span></div>
  <div class="infoContainer-infoItem"><span class="infoItem-header">Staked</span><span class="infoItem-value"> 0</span></div>
  <div class="infoContainer-infoItem"><span class="infoItem-header">APY</span><span class="infoItem-value"> </span></div>
</div>
`;
